R/utilities.R

```diff
@@ -94,10 +99,12 @@ prepareMachineLearningData <- function(formula, data, subset, subset.description
 
     # Treatment of missing values.
     # rebuild formula because there are new variables for each level of factorial predictors
-    predictor.names <- colnames(data)[colnames(data) != outcome.name]
-    form <- as.formula(paste(outcome.name, "~",
-                             paste(predictor.names[predictor.names != outcome.name], collapse = "+")))
-    processed.data <- EstimationData(form, data, subset, weights, missing, seed = seed)
+    if (dummy) {
```

Comment:
I don't understand why these lines are conditional now on "dummy" but were not before.

Reply:
redundant if no dummy variables